src/refract/JSONSchemaVisitor.cc

```diff
@@ -216,14 +231,67 @@ namespace refract
         }
     }
 
+    ArrayElement* JSONSchemaVisitor::arrayFromProps(std::vector<MemberElement*>& props)
+    {
+        ArrayElement *a = new ArrayElement;
+        a->renderType(IElement::rCompact);
+        for(std::vector<MemberElement *>::const_iterator i = props.begin();
+            i != props.end();
+            ++i) {
+            ObjectElement *ob = new ObjectElement;
+            ob->renderType(IElement::rCompact);
+            StringElement *str = TypeQueryVisitor::as<StringElement>((*i)->value.first);
+            JSONSchemaVisitor renderer(pDefs, fixed);
+            renderer.visit(*(*i)->value.second);
+            addDefinition(str->value, renderer.getOwnership());
+            StringElement *s = new StringElement;
```

Comment:
This didn't work for me so I just made it work, but will have a look into it, what was the problem.